### PR TITLE
7. FIRCD - replace install ID by heartbeat.

### PR DESCRIPTION
**Pull request by maksymmalyhin**
A next PR for CoreDiagnostic:
- heartbeat logic and tests ported
- install ID logic removed

### CODE REVIEW

File: Example/CoreDiagnostics/Tests/FIRCoreDiagnosticsTest.m [hunk 1]
```diff
@@ -31,10 +33,11 @@
 
 #import "FIRCDLibrary/Protogen/nanopb/firebasecore.nanopb.h"
 
+#import "FIRDiagnosticsDateFileStorage.h"
```
Comment (mikehaney24) [inline Example/CoreDiagnostics/Tests/FIRCoreDiagnosticsTest.m]: Use FIRCDLibrary-relative paths. This makes copybara easier.

File: Example/CoreDiagnostics/Tests/FIRCoreDiagnosticsTest.m [hunk 2]
```diff
@@ -77,6 +89,7 @@ extern void FIRPopulateProtoWithInfoPlistValues(
 
 // FIRCoreDiagnosticsInterop.
 + (void)sendDiagnosticsData:(nonnull id<FIRCoreDiagnosticsData>)diagnosticsData;
+- (void)sendDiagnosticsData:(nonnull id<FIRCoreDiagnosticsData>)diagnosticsData;
```
Comment (mikehaney24) [inline Example/CoreDiagnostics/Tests/FIRCoreDiagnosticsTest.m]: This method comes from FIRCoreDiagnosticsInterop, that also needs to be changed.

File: Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m [hunk 3]
```diff
@@ -33,6 +33,8 @@
 
 #import "FIRCDLibrary/Protogen/nanopb/firebasecore.nanopb.h"
 
+#import "FIRDiagnosticsDateFileStorage.h"
```
Comment (mikehaney24) [inline Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m]: Please use FIRCDLibrary-relative paths.

File: Example/CoreDiagnostics/Tests/FIRCoreDiagnosticsTest.m (hunk 1 above)
Comment (maksymmalyhin) [reply Example/CoreDiagnostics/Tests/FIRCoreDiagnosticsTest.m]: Updated

File: Example/CoreDiagnostics/Tests/FIRCoreDiagnosticsTest.m (hunk 2 above)
Comment (maksymmalyhin) [reply Example/CoreDiagnostics/Tests/FIRCoreDiagnosticsTest.m]: Excuse me, not sure I got it... What needs to be changed?

Comment (mikehaney24) [reply Example/CoreDiagnostics/Tests/FIRCoreDiagnosticsTest.m]: Sure! +sendDiagnosticsData is a method declared in `Interop/CoreDiagnostics/Public/FIRCoreDiagnosticsInterop.h`. You need to change the method declaration in that interop header (and how's it's called from FIRApp.m), or revert this change and change the implementation to use the +sharedInstance method. The latter would probably be easier.

Comment (maksymmalyhin) [reply Example/CoreDiagnostics/Tests/FIRCoreDiagnosticsTest.m]: Right now the instance method is private and is used by the public static method, so everything will be working as before. Would you like to get rid of the static method and use `sharedInstance` instead?

Comment (mikehaney24) [reply Example/CoreDiagnostics/Tests/FIRCoreDiagnosticsTest.m]: Got it--the diff can be hard to read with all the +s and -s. This part LGTM.

File: Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m [hunk 4]
```diff
@@ -149,66 +158,30 @@ + (instancetype)sharedInstance {
...
+                                                       transformers:nil
+                                                             target:kGDTTargetCCT];
+
+  FIRCoreDiagnosticsDateFileStorage *dateStorage = [[FIRCoreDiagnosticsDateFileStorage alloc]
+      initWithFileURL:[[self class] filePathURLWithName:kFIRDiagnosticsHeartbeatDateFileName]];
+
+  return [self initWithTransport:transport heartbeatDateStorage:dateStorage];
+}
+
+/** Initializer for unit tests */
```
Comment (mikehaney24) [inline Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m]: Please fix this docstring--periods after sentences, two missing ATparams, one missing ATreturn.

Comment (maksymmalyhin) [reply Firebase/CoreDiagnostics/FIRCDLibrary/FIRCoreDiagnostics.m]: Updated the doc string

**Pull request merged**